tests/integration/records/relationship-changes-test.js

```diff
@@ -949,3 +950,112 @@ test('Calling push with same belongsTo relationship does not trigger observer',
 
   assert.equal(observerCount, 0, 'author observer should not be triggered');
 });
+
+test('Calling push with hasMany relationship does trigger relation observer with already loaded objects', function(assert) {
```

Comment:
In the same file, there are 3 other tests related to belongsTo, testing the same behavior and they are named:
```js
test('Calling push with same belongsTo relationship does not trigger observer', function(assert) {
test('Calling push with updated belongsTo relationship trigger observer', function(assert) {
test('Calling push with relationship triggers willChange and didChange with detail when replacing different length in middle', function(assert) {
```

How should be named my tests according to your proposal?